pkg.nix

```diff
@@ -0,0 +1,173 @@
+# NOTE: This file is supposed to be similar to what is in Nixpkgs, except for
+# the `version`, `src` and `libtexpdf-src` attributes that are given by the
+# `flake.nix`. In Nixpkgs, we don't need `libtexpdf-src` because we use
+# `fetchFromGitHub` with fetchSubmodules = true;`.
+{ lib
+, stdenv
+, version, src, libtexpdf-src
+, autoreconfHook
+, gitMinimal
+, pkg-config
+, jq
+, cargo
+, rustc
+, rustPlatform
+, makeWrapper
+, poppler_utils
+, harfbuzz
+, icu
+, fontconfig
+, lua
+, libiconv
+, darwin
+, makeFontsConf
+, gentium
+, runCommand
+}:
+
+let
+  luaEnv = lua.withPackages(ps: with ps; [
+    cassowary
+    cldr
+    cosmo
+    fluent
+    linenoise
+    loadkit
+    lpeg
+    lua-zlib
+    lua_cliargs
+    luaepnf
+    luaexpat
+    luafilesystem
+    luarepl
+    luasec
+    luasocket
+    luautf8
+    penlight
+    vstruct
+    # lua packages needed for testing
+    busted
+    luacheck
+  # NOTE: Add lua packages here, to change the luaEnv also read by `flake.nix`
+  ] ++ lib.optionals (lib.versionOlder lua.luaversion "5.2") [
+    bit32
+  ] ++ lib.optionals (lib.versionOlder lua.luaversion "5.3") [
+    compat53
+  ]);
+in stdenv.mkDerivation (finalAttrs: {
+  pname = "sile";
+  inherit version src;
+
+  preAutoreconf = ''
+    # Add the libtexpdf src instead of the git submodule. (From some reason
+    # without --no-preserve=mode flag, libtexpdf/ is unwriteable). As explained
+    # before, in Nixpkgs, we won't need to run these commands.
+    rm -rf ./libtexpdf
+    cp --no-preserve=mode -r ${libtexpdf-src} ./libtexpdf/
+    # pretend to be a tarball release so sile --version will not say `vUNKNOWN`.
+    echo ${finalAttrs.version} > .tarball-version
+  '';
+
+  nativeBuildInputs = [
+    autoreconfHook
+    gitMinimal
+    pkg-config
+    jq
+    cargo
+    rustc
+    rustPlatform.cargoSetupHook
+    poppler_utils
+    makeWrapper
+  ];
+  # In Nixpkgs, we don't copy the Cargo.lock file from the repo to Nixpkgs'
+  # repo, but we inherit src, and specify a hash (it is a fixed output
+  # derivation). `nix-update` and `nixpkgs-update` should be able to catch that
+  # hash and update it as well when performing updates.
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = ./Cargo.lock;
+  };
+
+  buildInputs = [
+    luaEnv
+    harfbuzz
+    icu
+    fontconfig
+    libiconv
+  ] ++ lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.AppKit
+  ];
+
+  configureFlags = [
+    "--with-system-luarocks"
+    # TODO: Explain this flag
+    "PDFINFO=false"
+    #"--with-manual" In Nixpkgs we add this flag, here its not important enough
+  ];
+
+  postPatch = ''
+    patchShebangs build-aux/*.sh
+  '' + lib.optionalString stdenv.isDarwin ''
+    sed -i -e 's|@import AppKit;|#import <AppKit/AppKit.h>|' src/macfonts.m
```

Comment:
Not necessarily related to this change, but I consider such `sed` hooks rather ugly, and usually they are a result of upstream developers who don't make efforts for their distributors. This is not the case with sile, as development here is very much involved with distribution, so I was wondering whether in the `develop` branch you would like to make the effort to fix this issue, or at least explain better why is it needed only for Nix builds. Note it is rarely needed in Nixpkgs from my experience.

Reply:
Sure, I'll look into it. I do know the only other please we frequently build for macOS is on Homebrew which works properly with the current setup, hence my being reluctant to hack on it.

Reply:
https://github.com/sile-typesetter/sile/issues/1879